core/refresh_cache.go

```
@@ -0,0 +1,493 @@
+	PlacementType string `boil:"placement_type" json:"placement_type" toml:"placement_type" yaml:"placement_type"`
+}
+
+type RefreshCacheSlice []*RefreshCache
+
+type RefreshCacheRealtimeRecord struct {
+	Rule         string `json:"rule"`
+	RefreshCache int16  `json:"refresh_cache"`
+	RuleID       string `json:"rule_id"`
+}
+
+type GetRefreshCacheOptions struct {
+	Filter     RefreshCacheFilter     `json:"filter"`
+	Pagination *pagination.Pagination `json:"pagination"`
+	Order      order.Sort             `json:"order"`
+	Selector   string                 `json:"selector"`
+}
+
+type RefreshCacheFilter struct {
+	Publisher filter.StringArrayFilter `json:"publisher,omitempty"`
+	Domain    filter.StringArrayFilter `json:"domain,omitempty"`
+	Country   filter.StringArrayFilter `json:"country,omitempty"`
+	Device    filter.StringArrayFilter `json:"device,omitempty"`
+	Active    filter.StringArrayFilter `json:"active,omitempty"`
+}
+
+func (rc *RefreshCache) FromModel(mod *models.RefreshCache) error {
+	rc.RuleId = mod.RuleID
+	rc.Publisher = mod.Publisher
+	rc.Domain = mod.Domain
+	rc.RefreshCache = mod.RefreshCache
+	rc.RuleId = mod.RuleID
+
+	if mod.Os.Valid {
+		rc.OS = mod.Os.String
+	}
+
+	if mod.Country.Valid {
+		rc.Country = mod.Country.String
+	}
+
+	if mod.Device.Valid {
+		rc.Device = mod.Device.String
+	}
+
+	if mod.PlacementType.Valid {
+		rc.PlacementType = mod.PlacementType.String
+	}
+
+	if mod.Browser.Valid {
+		rc.Browser = mod.Browser.String
+	}
+
+	return nil
+}
+
+func (r *RefreshCacheService) CreateRefreshCache(ctx context.Context, data *dto.RefreshCacheUpdateRequest) error {
+
+	rc := RefreshCache{
+		Publisher:     data.Publisher,
+		Domain:        data.Domain,
+		Country:       data.Country,
+		Device:        data.Device,
+		RefreshCache:  data.RefreshCache,
+		Browser:       data.Browser,
+		OS:            data.OS,
+		PlacementType: data.PlacementType,
+	}
+
+	mod := rc.ToModel()
+
+	old, err := r.prepareHistory(ctx, mod)
```

Error is ignored. Also we can't have `old` when we haven't created `new` yet.

---

^^^